lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -2087,6 +2087,9 @@ void FunctionEntrypointAnalyzer::runInterproceduralAnalysis() {
 
 template<bool ShouldAnalyzeABI>
 bool EarlyFunctionAnalysis<ShouldAnalyzeABI>::runOnModule(Module &M) {
+  if (not M.getFunction("root") or M.getFunction("root")->isDeclaration())
+    return false;
```

Comment:
* Are you sure we're not supposed to `assert` here?
* You seem to be using the `not getFunction(x) or M.getFunction(x)->isDeclaration()` quite a bit. Make an helper function in `IRHelpers.h`.

Reply:
What scenario could trigger this?
If one asks for the binary at the end of step that includes `collect-cfg` and therefore we run the step but `cloneFilter`ing only for things we're interested in, which means, only the binary and therefore we have no `root`?

Reply:
yes